QuickStart/ViewController.swift

```diff
@@ -0,0 +1,320 @@
+                self.updateLogging(text: "Could not acquire token: No result returned")
+                return
+            }
+
+            if (result.status != AD_SUCCEEDED) {
+
+                if result.error.domain == ADAuthenticationErrorDomain
+                    && result.error.code == ADErrorCode.ERROR_UNEXPECTED.rawValue {
+                    
+                    self.updateLogging(text: "Unexpected internal error occured");
+                    
+                } else {
+                    
+                    self.updateLogging(text: result.error.description)
+                }
+                
+                return
+            }
+            
+            self.updateLogging(text: "Access token is \(String(describing: result.accessToken))")
+            self.updateSignoutButton(enabled: true)
+            completion(true)
+        }
+    }
+
+    func acquireTokenSilently(completion: @escaping (_ success: Bool) -> Void) {
+
+        guard let applicationContext = self.applicationContext else { return }
+        
+        /**
+
+         Acquire a token for an existing account silently
+         - withResource:        The resource you wish to access. This will the Microsoft Graph API for this sample.
+         - clientId:            The clientID of your application, you should get this from the app portal.
+         - redirectUri:         The redirect URI that your application will listen for to get a response of the
+                                Auth code after authentication. Since this a native application where authentication
+                                happens inside the app itself, we can listen on a custom URI that the SDK knows to
+                                look for from within the application process doing authentication.
+         - completionBlock:     The completion block that will be called when the authentication
+                                flow completes, or encounters an error.
+         
+         */
+        
+        
+
+        applicationContext.acquireTokenSilent(withResource: kGraphURI, clientId: kClientID, redirectUri:kRedirectUri) { (result) in
+            
+            guard let result = result else {
+                
+                self.updateLogging(text: "Could not acquire token: No result returned")
+                return
+            }
+
+           if (result.status != AD_SUCCEEDED) {
+
+                // USER_INPUT_NEEDED means we need to ask the user to sign-in. This usually happens
+                // when the user's Refresh Token is expired or if the user has changed their password
+                // among other possible reasons.
+
+            if result.error.domain == ADAuthenticationErrorDomain
+                && result.error.code == ADErrorCode.ERROR_SERVER_USER_INPUT_NEEDED.rawValue {
+                
+                    DispatchQueue.main.async {
+                        self.acquireToken() { (success) -> Void in
+                            if success {
```

Comment:
Should completion be called in here? Otherwise, there's no handling of success in here.